eth-contracts/contracts/staking/Staking.sol

```diff
@@ -2,17 +2,18 @@ pragma solidity ^0.5.0;
 
 import "./ERCStaking.sol";
 import "./Checkpointing.sol";
-
-import "./res/Autopetrified.sol";
-import "./res/IsContract.sol";
-
 import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 import "openzeppelin-solidity/contracts/token/ERC20/SafeERC20.sol";
 import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
 import "openzeppelin-solidity/contracts/token/ERC20/ERC20Burnable.sol";
+import "../res/IsContract.sol";
+import "../res/openzeppelin/Initializable.sol";
+import "../res/TimeHelpers.sol";
+import "../service/registry/RegistryContract.sol";
 
 
-contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IsContract {
+/** NOTE - will call RegistryContract.constructor */
+contract Staking is Initializable, RegistryContract, ERCStaking, ERCStakingHistory, IsContract, TimeHelpers {
```

Comment:
is the registry contract constructor called on deployment of the logic contract...?

Reply:
added TODO to figure out Ownable() vs proxy logic after merge